modules/apps/healthcheck/healthcheck-bestpractice/src/main/java/com/liferay/healthcheck/bestpractice/internal/DefaultSiteInitializerUserHealthcheck.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.healthcheck.bestpractice.internal;
+
+import com.liferay.healthcheck.Healthcheck;
+import com.liferay.healthcheck.HealthcheckItem;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.portal.kernel.exception.NoSuchUserException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Several SiteInitializers create default users - once such a SiteInitializer
+ * has been used, those might be present in production systems. Check for
+ * their existence and trigger admins to delete them or ignore, if this is a
+ * dev system, and if it's acceptable to have an account where _everybody_
+ * can reset the password (e.g. through mailinator.com).
+ *
+ * @author Olaf Kock
+ */
+@Component(service = Healthcheck.class)
+public class DefaultSiteInitializerUserHealthcheck implements Healthcheck {
+
+	@Override
+	public Collection<HealthcheckItem> check(long companyId) {
+		Collection<HealthcheckItem> result = new LinkedList<>();
+
+		for (String user : _KNOWN_USERS) {
+			_checkForUser(companyId, result, user);
+		}
+
+		if (result.isEmpty()) {
+			result.add(
+				new HealthcheckItem(true, getClass().getName(), null, _MSG));
+		}
+
+		return result;
+	}
+
+	@Override
+	public String getCategory() {
+		return "healthcheck-category-bestpractice";
+	}
+
+	private void _checkForUser(
+		long companyId, Collection<HealthcheckItem> result,
+		String mailAddress) {
+
+		try {
+			User user = _userLocalService.getUserByEmailAddress(
+				companyId, mailAddress);
+
+			if (user != null) {
+				String link = new StringBundler(
+					_LINK
+				).append(
+					_LINK_PARAMETER
+				).append(
+					user.getUserId()
+				).toString();
+
+				result.add(
+					new HealthcheckItem(
+						false,
+						StringUtil.merge(
+							new String[] {getClass().getName(), mailAddress},
+							"-"),
+						link, _MSG_FOUND, mailAddress));
+			}
+		}
+		catch (PortalException portalException) {
+			if (!(portalException instanceof NoSuchUserException)) {
+				result.add(new HealthcheckItem(this, portalException));
```

Comment:
I figured I'd like to be expressive. I'm not aware what exceptions might be thrown when PortalException is declared and I wanted to be explicit in _only_ handling this one, with no thought about what else might be thrown today or tomorrow. 
Personally, I'd like to stay this explicit, unless this is critical for you. If this wasn't used within a loop, and for only a single user, I'd actually prefer to just throw any uncaught exception.

Should I change this in tomorrow's push?

Reply:
This try block is incapable of throwing PortalException other than NoSuchUserException, so this if check is essentially dead code. But my main concern is on this exception version constructor of HealthcheckItem. I doubt it is needed, at least for now this is the only usage of it. I'd rather remove it to keep HealthcheckItem simple. If this type of constructor turns out to be needed later, we can always re-add.

Reply:
got it... key was actually to call fetchUserByEmailAddress instead of getUserByEmailAddress, to eliminate all exceptions thrown by _userLocalService. Simplified, force-pushed the new branch.